src/resource/file.ts

```diff
@@ -0,0 +1,157 @@
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import {
+  UrlString,
+  getFile as coreGetFile,
+  overwriteFile as coreOverwriteFile,
+  saveFileInContainer as coreSaveFileInContainer,
+} from "@inrupt/solid-client";
+import { VerifiableCredential } from "@inrupt/solid-client-vc";
+import { fetchWithVc } from "../fetch";
+import { FetchOptions } from "../type/FetchOptions";
+
+/**
+ * Retrieve a File from a Solid Pod using an Access Grant to prove the caller
+ * is authorized to access the target resource.
+ *
+ * @param resourceUrl The URL of the target resource.
+ * @param accessGrant The Access Grant VC proving the caller is authorized.
+ * @param options Optional properties to customise the request behaviour.
+ * @returns A promise that resolves to a File if successful, and that rejects otherwise.
+ * @since 0.4.0
+ */
+export async function getFile(
+  resourceUrl: UrlString,
+  accessGrant: VerifiableCredential,
+  options?: FetchOptions
+): ReturnType<typeof coreGetFile> {
+  const fetchOptions: FetchOptions = {};
+  if (options && options.fetch) {
+    fetchOptions.fetch = options.fetch;
+  }
+
+  const authenticatedFetch = await fetchWithVc(
+    resourceUrl,
+    accessGrant,
+    fetchOptions
+  );
+
+  return coreGetFile.apply(undefined, [
+    resourceUrl,
+    {
+      fetch: authenticatedFetch,
+    },
+  ]);
+}
+
+/**
+ * Saves a file at a given URL using an Access Grant to prove the caller is
+ * authorized to save a file at the given URL. If a file already exists in the
+ * at the URL, the function overwrites the existing file.
+ *
+ * @param resourceUrl The URL where the file is to be saved.
+ * @param file The file to be written.
+ * @param accessGrant The Access Grant VC proving the caller is authorized.
+ * @param options Optional properties to customise the request behaviour, or override the Content-Type of the file.
+ * @returns A promise that resolves to a File if successful, and that rejects
+ * otherwise.
+ * @since unreleased
+ */
+export async function overwriteFile(
+  resourceUrl: UrlString,
+  file: File | Buffer,
+  accessGrant: VerifiableCredential,
+  options?: FetchOptions & { contentType?: string }
+): ReturnType<typeof coreOverwriteFile> {
+  const fetchOptions: FetchOptions = {};
+  if (options && options.fetch) {
+    fetchOptions.fetch = options.fetch;
+  }
+
+  const authenticatedFetch = await fetchWithVc(
+    resourceUrl,
+    accessGrant,
+    fetchOptions
+  );
+
+  const overwriteFileOptions: { contentType?: string } = {};
+  if (options && options.contentType) {
+    overwriteFileOptions.contentType = options.contentType;
+  }
+
+  return coreOverwriteFile.apply(undefined, [
+    resourceUrl,
+    file,
+    {
+      ...overwriteFileOptions,
+      fetch: authenticatedFetch,
+    },
+  ]);
```

Comment:
Same comment as `fetchOptions` for `overwriteFileOptions`.
Same comment for `.apply`